ripgrep_required_reason_api_text_scanner.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+excluded_dirs=() # e.g. ("Pods" "3rdparty")
+search_string=(
+    # File timestamp APIs
+    # https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393
+    "creationDate"
+    "creationDateKey"
+    "modificationDate"
+    "fileModificationDate" 
+    "contentModificationDateKey" 
+    "creationDateKey"
+    "getattrlist\(" 
+    "getattrlistbulk\(" 
+    "fgetattrlist\(" 
+    "stat\(" 
+    "stat.st_" # see https://developer.apple.com/documentation/kernel/stat
+    "fstat\("
+    "fstatat\("
+    "lstat\("
+    "getattrlistat\("
+
+    # System boot time APIs
+    # https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278394
+    "systemUptime"
+    "mach_absolute_time\(\)"
+
+    # Disk space APIs
+    "volumeAvailableCapacityKey"
+    "volumeAvailableCapacityForImportantUsageKey"
+    "volumeAvailableCapacityForOpportunisticUsageKey"
+    "volumeTotalCapacityKey"
+    "systemFreeSize"
+    "systemSize"
+    "statfs\("
+    "statvfs\("
+    "fstatfs\("
+    "fstatvfs\("
+    "getattrlist\("
+    "fgetattrlist\("
+    "getattrlistat\("
+
+    # Active keyboard APIs
+    # https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278400
+    "activeInputModes"
+
+    # User defaults APIs
+    # https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401
+    "^UserDefaults"
```

Comment:
This is missing uses of UserDefaults in our app, presumably an erroneous ^.

If I remove the `^` it works fine!

Reply:
@simonmitchell can you give me some specifics / examples? I intentionally added the `^` to avoid some false positives such as `RCTDevSettingsUserDefaults`, `FlipperKitUserDefaultsPlugin`, and some things in Expo packages.